systest build wrongly includes cms.h for LibreSSL

~~The `build.rs` for `openssl-sys` correctly finds LibreSSL on FreeBSD thanks `pkg-config`. The `build.rs` for `systest`, however, does not use `pkg-config` at all, so it gets to read system OpenSSL headers from `/usr` instead of package-installed LibreSSL headers from `/usr/local`:~~

```
error: failed to run custom build command for `systest v0.1.0 (file:///usr/home/greg/src/github.com/sfackler/rust-openssl/systest)`
process didn't exit successfully: `/usr/home/greg/src/github.com/sfackler/rust-openssl/target/debug/build/systest-6b98ceb05f24ce15/build-script-build` (exit code: 101)
--- stdout
-----------------------------------------
[…]
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-I" "/usr/local/include" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-o" "/usr/home/greg/src/github.com/sfackler/rust-openssl/target/debug/build/systest-94a08862ec32b47d/out/all.o" "-c" "/usr/home/greg/src/github.com/sfackler/rust-openssl/target/debug/build/systest-94a08862ec32b47d/out/all.c"
cargo:warning=In file included from /usr/home/greg/src/github.com/sfackler/rust-openssl/target/debug/build/systest-94a08862ec32b47d/out/all.c:19:
cargo:warning=/usr/include/openssl/cms.h:61:4: error: CMS is disabled.
cargo:warning=#  error CMS is disabled.
cargo:warning=   ^
cargo:warning=/usr/include/openssl/cms.h:193:52: error: unknown type name 'ossl_ssize_t'; did you mean '__ssize_t'?
cargo:warning=                              unsigned char *pass, ossl_ssize_t passlen);
cargo:warning=                                                   ^
[…]
```

hmm. It actually does pass `-I /usr/local/include`, but `/usr/include/openssl/cms.h` gets included anyway — because LibreSSL does not have `cms.h` I think

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -35,6 +35,7 @@ job: &JOB
   steps:
   - checkout
   - run: apt-get update
+  - run: apt-get remove -y libssl-dev
   - run: ./test/add_target.sh
   - *RESTORE_REGISTRY
   - run: cargo generate-lockfile

--- a/systest/build.rs
+++ b/systest/build.rs
@@ -5,6 +5,7 @@ use std::env;
 fn main() {
     let mut cfg = ctest::TestGenerator::new();
     let target = env::var("TARGET").unwrap();
+    let mut is_libressl = false;
 
     if let Ok(out) = env::var("DEP_OPENSSL_INCLUDE") {
         cfg.include(&out);
@@ -24,6 +25,7 @@ fn main() {
 
     if let Ok(_) = env::var("DEP_OPENSSL_LIBRESSL") {
         cfg.cfg("libressl", None);
+        is_libressl = true;
     } else if let Ok(version) = env::var("DEP_OPENSSL_VERSION") {
         cfg.cfg(&format!("ossl{}", version), None);
     }
@@ -40,12 +42,6 @@ fn main() {
             cfg.cfg("osslconf", Some(var));
         }
     }
-
-    let has_cms_h = if let Ok(version) = env::var("DEP_OPENSSL_LIBRESSL_VERSION") {
-        version != "261" && version != "262"
-    } else {
-        true
-    };
 
     cfg.header("openssl/comp.h")
         .header("openssl/dh.h")
@@ -64,7 +60,7 @@ fn main() {
         .header("openssl/aes.h")
         .header("openssl/ocsp.h");
 
-    if has_cms_h {
+    if !is_libressl {
         cfg.header("openssl/cms.h");
     }
 

